src/inspector.rs

```diff
@@ -278,6 +279,8 @@ impl ChannelBase {
   {
     let buf = std::mem::MaybeUninit::<T>::uninit();
     let embedder_ptr: *const T = buf.as_ptr();
+    // TODO: the call to base() creates a reference to uninitialized memory (UB)
+    // fixing this requires changes in the ChannelImpl trait, namely ChannelImpl::base() can't take &self
```

Comment:
Could this possible be addressed in this PR? If not please add your handle to `TODO()`

Reply:
I initially tried addressing it in this PR, but I realized that changing `&self` to something like `this: *const Self` (to avoid the reference to uninitialized memory) isn't as trivial as I thought because it propagates to other types and also affects the public API. I am also not entirely sure if that's the right approach to that, so I thought that deserves its own PR.

Reply:
I created #958 for tracking this TODO item and ideas for a potential fix